Managers/UInputMan.cpp

```diff
@@ -837,8 +839,19 @@ namespace RTE {
 			return;
 		}
 
-		if (g_InActivity) {
-			if (AnyStartPress(false)) {
+		if (g_InActivity) {	
+			bool BuyMenuIsOpen = 0;
+			GameActivity* temp = dynamic_cast<GameActivity*>(g_ActivityMan.GetActivity());
+
+			// Make ESC key open BuyMenuGUIs if any are open rather than pausing the game.
+			for (short player = Players::PlayerOne; player < Players::MaxPlayerCount; player++) {
```

Comment:
Rather than max player count, it'd be better to use the current activity's player count with Activity::GetPlayerCount